File: app-aws-lambda-example-go/logic/AppExampleCommandSet.go
package logic

import (
	ccmd "github.com/pip-services3-go/pip-services3-commons-go/commands"
	cconv "github.com/pip-services3-go/pip-services3-commons-go/convert"
	"github.com/pip-services3-go/pip-services3-commons-go/run"
	cvalid "github.com/pip-services3-go/pip-services3-commons-go/validate"
)

type AppExampleCommandSet struct {
	ccmd.CommandSet
	controller *AppExampleController
}

func NewAppExampleCommandSet() *AppExampleCommandSet {
	c := &AppExampleCommandSet{
		CommandSet: *ccmd.NewCommandSet(),
		controller: NewAppExampleController(),
	}

	return c
}

func (c *AppExampleCommandSet) greetingCommand() ccmd.ICommand {
	return ccmd.NewCommand(
		"greeting",
		cvalid.NewObjectSchema().
			WithRequiredProperty("name", cconv.String),
		func(correlationId string, args *run.Parameters) (interface{}, error) {
			name := args.GetAsString("name")
			return c.controller.Greeting(name)
		},
	)
}

File: app-container-example-nodex/src/logic/AppExampleCommandSet.ts
import { CommandSet } from 'pip-services3-commons-nodex';
import { Command } from 'pip-services3-commons-nodex';
import { ObjectSchema } from 'pip-services3-commons-nodex';
import { ICommand } from 'pip-services3-commons-nodex';
import { TypeCode } from 'pip-services3-commons-nodex';
import { Parameters } from 'pip-services3-commons-nodex';
import { AppExampleController } from './AppExampleController';


export class AppExampleCommandSet extends CommandSet {
    private _logic: AppExampleController

    constructor(logic: AppExampleController) {
        super();
        this._logic = logic;

        this.addCommand(this.greetingCommand());
    }

    private greetingCommand(): ICommand {
        return new Command(
            'greeting',
            new ObjectSchema(true).withOptionalProperty('name', TypeCode.String),
            (correlationId: string, args: Parameters): Promise<string> => {
                const name = args.getAsString('name');
                return this._logic.greeting(name);
            }
        );
    }
}